CMSgov/managed-care-review
# MR-2528: Multi-rate certs show up in emails

## Summary
[MR-2528](https://qmacbis.atlassian.net/browse/MR-2528)

- Refactor emailer functions for multi-rate packages.
- Refactor utility functions for multi-rate packages.
- Refactor mock data for multi-rate packages.
- Fixed existing bugs in emailer function tests.
- Added `Prerequisites` section in Launch Darkly Testing technical doc `launch-darkly-testing-approach.md`.

#### Related issues

#### Screenshots

#### Test cases covered

**Note**: Various tests have been refactored and updated for the change in the domain model for multi-rates. These were missed with my previous PR.

**Emailer Tests**
- FIXES: There was some bugs in many of the emailer tests where programIDs and rateProgramIDs were not found in the state program lists for the specific state. The tests still passed and only errors were logged.
- `newPackageCMSEmail.test.ts`
   - `'includes expected data summary for a multi-rate contract and rates submission CMS email'`
- `newPackageStateEmail.test.ts`
   - `'includes expected data summary for a multi-rate contract and rates submission State email'`
- `resubmitPackageCMSEmail.test.ts`
   - `'includes expected data summary for a contract and rates resubmission CMS email'`
   - `'includes expected data summary for a multi-rate contract and rates resubmission CMS email'`
- `resubmitPackageStateEmail.test.ts`
   - `'includes expected data summary for a contract and rates resubmission State email'`
   - `'includes expected data summary for a multi-rate contract and rates resubmission State email'`
- `unlockPackageCMSEmail.test.ts`
   - `'includes expected data summary for a contract and rates submission unlock CMS email'`
   - `'includes expected data summary for a multi-rate contract and rates submission unlock CMS email'`
- `unlockPackageStateEmail.test.ts`
   - `'includes expected data summary for a contract and rates submission unlock State email'`
   - `'includes expected data summary for a multi-rate contract and rates submission unlock State email'`



## QA guidance

[Figma - State Emails](https://www.figma.com/file/frKNnm6lkcpdjJ2eVJE8Bj/Managed-Care-Review?node-id=107%3A16026)
[Figma - CMS Emails](https://www.figma.com/file/frKNnm6lkcpdjJ2eVJE8Bj/Managed-Care-Review?node-id=106%3A14437)
Email design for single rate and multi rate differ on `unlock` and `resubmit` emails. Here are screenshots of the difference since it is not on Figma.

![Screen Shot 2022-10-06 at 3 09 22 PM](https://user-images.githubusercontent.com/98117700/194398876-c8063c43-412a-4b3d-83d2-265c4fc8670d.png)
![Screen Shot 2022-10-06 at 3 04 48 PM](https://user-images.githubusercontent.com/98117700/194398880-54bae76a-9ee4-4115-b0c1-adf54acc6ac8.png)


All emails have been updated to accept multi-rate submissions. Please make sure to check:
- [x] New Submission CMS Emails
   - [x] Single Rate submission
   - [x] Multi Rate submission
- [x] New Submission State Emails
   - [ ] Single Rate submission
   - [ ] Multi Rate Submission
- [x] Submission unlock Emails
   - [ ] Single Rate submission unlock
   - [x] Multi Rate Submission unlock
- [x] Resubmission Emails
   - [x] Single Rate resubmission
   - [x] Multi Rate resubmission
- [x] Changes in were made to `generateRateName()` please double check rate names for each rate certification.
